tests/integration/multicluster/cluster_local.go

```diff
@@ -24,27 +24,55 @@ import (
 	"istio.io/istio/pkg/test/framework/components/namespace"
 	"istio.io/istio/pkg/test/framework/components/pilot"
 	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
 )
 
 // ClusterLocalTest tests that traffic works within a local cluster while in a multicluster configuration
+// clusterLocalNS have been configured in meshConfig.serviceSettings to be clusterLocal.
 func ClusterLocalTest(t *testing.T, clusterLocalNS namespace.Instance, pilots []pilot.Instance) {
 	framework.NewTest(t).
 		Label(label.Multicluster).
 		Run(func(ctx framework.TestContext) {
-			cluster1 := ctx.Environment().Clusters()[0]
-			cluster2 := ctx.Environment().Clusters()[1]
+			ctx.NewSubTest("respect-cluster-local-config").Run(func(ctx framework.TestContext) {
+				clusters := ctx.Environment().Clusters()
+				for i := range clusters {
+					i := i
+					ctx.NewSubTest(fmt.Sprintf("cluster-%d cluster local", i)).
+						RunParallel(func(ctx framework.TestContext) {
+							local := clusters[i]
+							remotes := getRemoteClusters(clusters, i)
```

Comment:
let's not use the term `remote` here as that might be confused with master/remote. I actually wonder if we need `getRemoteClusters` at all ... I think you could just iterate over all clusters and skip `i`.